community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/labelscan/LuceneLabelScanStore.java

```diff
@@ -114,9 +116,22 @@ else if ( !luceneIndex.isValid() )
         }
     }
 
+    private void create() throws IOException
```

Comment:
We currently already have `ReadOnlyAbstractDatabaseIndex/WritableAbstractDatabaseIndex` but those expect AbstractLuceneIndex maybe those can be updated to expect something else - but those are really lucene oriented currently so that can be some additional work; 
or instead we can just delegate creation to index always - in that case index will replace it's internal structures, not sure how that will work with current new index